spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/convert/ConverterAwareMappingSpannerEntityReader.java

```diff
@@ -75,8 +75,8 @@ public <R> R read(
     StructPropertyValueProvider propertyValueProvider =
         new StructPropertyValueProvider(structAccessor, this.converter, this, allowMissingColumns);
 
-    PreferredConstructor<?, SpannerPersistentProperty> persistenceConstructor =
-        persistentEntity.getPersistenceConstructor();
+    InstanceCreatorMetadata<SpannerPersistentProperty> instanceCreatorMetadata =
```

Comment:
I am probably missing context here, but can you explain what is the rational for replacing `PreferredConstructor` with `instanceCreatorMetadata`? Is there any upstream Spring changes that's related?

Reply:
[`PreferredConstructor.isConstructorParameter()`](https://github.com/spring-projects/spring-data-commons/blob/main/src/main/java/org/springframework/data/mapping/PreferredConstructor.java#L92)` is deprecated in Spring Data.

Reply:
ah, I missed that. Thank you.